tests/fuzz/runFuzzTest.sh

```diff
@@ -31,6 +31,9 @@ FUZZ_TIMEOUT_SEC=$2
 BUILD_DIR_PATH=$3
 CORPUS_UPLOAD_LOC=$4
 ARTIFACT_UPLOAD_LOC=$5
+FUZZ_COVERAGE=$6
+S2N_ROOT=$7
```

Comment:
Are you using this variable anywhere?  It's sort of a historical holddover from the Makefiles... and between CodeBuild and CMake, there should be existing equivalents defined.

Reply:
Ya, 7 CLI arguments is a lot, and I worry about the maintainability of this.

How can we make things simpler?

### 1 - Remove Unnecessary Options
My understanding is that right now `BUILDER_DIR_PATH`, `CORPUS_UPLOAD_LOC`, and `ARTIFACT_UPLOAD_LOC` are never actually set to anything else. And I'd imagine that if they _were_ set to something else, a whole host of things would break. That's a good sign that these aren't actually configurable, and are probably better off being represented as constant values.

### 2 - Separation of Concerns
Generally it's best to "separate concerns". Don't unnecessarily couple things that don't need to be coupled. This applies to shell scripts that same way that it applies to actual code
```
// tricky and complex, probably not the best option
fn run_fuzz_tests_and_generate_coverage_report(arg1, arg2, arg3, arg4, arg5, arg6, arg6)

// a bit more manageable
fn run_fuzz_tests(arg1, arg2, arg3, arg4)
fn generate_coverage_report(arg5, arg6, arg7)
```

I'd imagine an ideal "coverage pipeline" to look something like this
1. build the fuzz test binaries with instrumentation enabled
2. download the corpus files
3. run the fuzz tests (which is mostly agnostic as to whether or not coverage is enabled)
4. upload the improved corpus files
5. merge the coverage information/generate the report
6. upload the report

It seems odd that all of the coverage reporting is living inside a script called `runFuzzTests`. Perhaps we could refactor [coverage_report.sh](https://github.com/aws/s2n-tls/blob/main/codebuild/bin/coverage_report.sh) to be slightly more generic, and then everything could use the same script?

### 3 - Scrutinize Coverage Assertions
I like the idea of asserting on some sort of coverage information. "feature coverage" from the fuzzer seems very nice and easy to assert on.

Asserting on coverage from the actual report seems to requires a big blob of `sed`/`grep`/`regex` all squished together, and that is not sparking joy. 

What benefit does this complicated stuff provide that feature coverage doesn't? And are our thresholds/assertions well-researched enough to actually be useful? I know we found that session ticket coverage issue when manually reviewing the code, so I question the value of these more complex assertions.

Reply:
I believe I should be able to remove `BUILD_DIR_PATH` and just give a hardcoded path to the build directory.
`CORPUS_UPLOAD_LOC`, and `ARTIFACT_UPLOAD_LOC`, are s3 bucket locations (For example, "s3:://bucketName"). There is probably no security concerns to make it public, as it requires correct IAM roles to be able to access the bucket. Is it fine making such resource info hard-coded const?

Reply:
I did some refactoring so it only takes 3 CLI arguments instead of 7. 

Since `runFuzzTest.sh` is very large and hard to read, I have broken it into smaller components which are:

runFuzzTest.sh: responsible for running the actual fuzz test
fuzz_corpus_download.sh: responsible for downloading corpus files from S3
fuzz_corpus_upload.sh: responsible for uploading improved corpus files to S3
fuzz_coverage_report.sh: responsible for generating coverage information

On top of that, I removed couple of print statements that were hidden by CTest. I left the outputs that are useful for debugging in case fuzzing fails.